map.go

```diff
@@ -14,6 +14,7 @@ import (
 // Errors returned by Map and MapIterator methods.
 var (
 	ErrKeyNotExist      = xerrors.New("key does not exist")
+	ErrKeyExist         = xerrors.New("key does exist")
```

Comment:
I'd suggest to slightly rephrase the error message:

```suggestion
	ErrKeyExist         = xerrors.New("key already exists")
```